otlp_json2/__init__.py

```diff
@@ -0,0 +1,173 @@
+from __future__ import annotations
+
+import json
+from collections.abc import Mapping, Sequence
+from typing import Any, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
+    from opentelemetry.trace import Link
+    from opentelemetry._logs import LogRecord
+    from opentelemetry.sdk.trace import ReadableSpan, Event
+    from opentelemetry.sdk.resources import Resource
+    from opentelemetry.sdk.util.instrumentation import InstrumentationScope
+    from opentelemetry.trace.status import Status
+
+    _LEAF_VALUE: TypeAlias = "str | int | float | bool"  # TODO: confirm
+    _VALUE: TypeAlias = "_LEAF_VALUE | Sequence[_LEAF_VALUE]"
+
+
+CONTENT_TYPE = "application/json"
+
+
+_VALUE_TYPES = {
+    # NOTE: order matters, for isinstance(True, int).
+    bool: ("boolValue", bool),
+    int: ("intValue", str),
```

Comment:
int values are presented to string representation in JSON format.
I think it's because JavaScript would coerce larger int values to double instead.
Maybe I should make this `lambda x: str(int(x))` to be both more explicit and work correctly if someone subclasses int.

Reply:
Got it, thanks. `str(x)` should work fine if someone subclasses int, no?

Reply:
But a line comment wouldn't hurt explaining why.